### PR TITLE
fix base fee check logic in state transition

### PR DESCRIPTION
**Pull request by yihuang**
## Description

- should check london hardfork first, otherwise it panic if feemarket not registered.

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Review by fedekunze — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -67,6 +67,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (evm) [tharsis#901](https://github.com/tharsis/ethermint/pull/901) Support multiple MsgEthereumTx in single tx.
 * (config) [tharsis#908](https://github.com/tharsis/ethermint/pull/908) Add api.enable flag for Cosmos SDK Rest server
 * (feemarket) [tharsis#919](https://github.com/tharsis/ethermint/pull/919) Initialize baseFee in default genesis state.
+* (evm) [tharsis#932](https://github.com/tharsis/ethermint/pull/932) Fix base fee check logic in state transition.
```
Comment (fedekunze) [inline CHANGELOG.md:70]: this should be in the bug fixes section?

Comment (yihuang) [reply CHANGELOG.md]: I thought it was not an issue for a new network, but yeah I guess it should be a bug fix, changed now.

File: x/evm/keeper/state_transition.go [hunk 2]
```diff
@@ -110,8 +113,8 @@ func (k Keeper) VMConfig(ctx sdk.Context, msg core.Message, params types.Params,
 		Debug:       debug,
 		Tracer:      tracer,
 		NoRecursion: false, // TODO: consider disabling recursion though params
-		NoBaseFee:   fmParams.NoBaseFee,
-		ExtraEips:   params.EIPs(),
+		NoBaseFee:   noBaseFee,
```
Comment (yihuang) [inline x/evm/keeper/state_transition.go:116]: Actually, I searched in go-ethereum's code, the only place this field is used is here: https://github.com/ethereum/go-ethereum/blob/master/core/state_transition.go#L238, which is not used by us.
So maybe we can hardcode a `true` here to save a query? @fedekunze

**Comment by fedekunze** [comment]
@yihuang can you fix the RPC test that's failing?

**Comment by yihuang** [comment]
> @yihuang can you fix the RPC test that's failing?

I just resolved conflicts and rpc test succeeds.

**Pull request merged**